src/pages/SignUp.jsx

```diff
@@ -0,0 +1,160 @@
+
+  const onMutate = (e) => {
+    //checks the type of input and sets the value of the input to the formData
+    if (e.target.files) {
+      setFormData((prev) => {
+        return {
+          ...prev,
+          images: e.target.files,
+        };
+      });
+    }
+    if (!e.target.files) {
+      setFormData((prev) => {
+        return {
+          ...prev,
+          [e.target.name]: e.target.value,
+        };
+      });
+    }
+  };
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const storeImage = async (image) => {
+      return new Promise((resolve, reject) => {
+        //creates reference to the storage bucket
+        const storageRef = ref(storage, "pictures/" + image.name);
+        //uploads the image to the storage bucket
+        const uploadTask = uploadBytesResumable(storageRef, image);
+        //listens for the upload to finish
+        uploadTask.on(
+          "state_changed",
+          (smapshot) => {
+            const progress =
+              (smapshot.bytesTransferred / smapshot.totalBytes) * 100;
+            console.log("Upload is " + progress + "% done");
+          },
+          (error) => {
+            reject(error);
+          },
+          () => {
+            //gets the download url of the image
+            getDownloadURL(uploadTask.snapshot.ref).then((downloadURL) => {
+              resolve(downloadURL);
+            });
+          }
+        );
+      });
+    };
+
+    //awaits the promise to store the image url
+    const imageUrls = await Promise.all(
+      //loops through the images array
+      [...images].map((image) => storeImage(image))
+    ).catch((err) => {
+      console.log(err);
+      return;
+    });
+    try {
+      //FireBase provides with the method of creating a user with email and password
+      //creates a new user with the email and password
+      const auth = getAuth();
+
+      const userCredentials = await createUserWithEmailAndPassword(
+        auth,
+        email,
+        password
+      );
+      //creates a new user document in the users collection
+      const user = userCredentials.user;
+      //creates a copy of the form data and removes the images property
+      const fromDataCopy = { ...formData, imageUrls };
+      //we are not going to store the state of images and password in the database
+      //we are going to store the image url in the database
+      //so we delete the images and password from the form data Copy and store it in the database
+      delete fromDataCopy.images;
+      delete fromDataCopy.password;
+      //sets the user document in the users collection and adds the form data
+      await setDoc(doc(db, "users", user.uid), fromDataCopy);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+  return (
+    <>
+      <div>
+        <h1>Signup</h1>
+        <Card className="data-form">
+          <h1>Please SignUp</h1>
+          <form className="data-form" onSubmit={handleSubmit}>
```

Comment:
shouldn't we have formik implemented here?

Reply:
Yes! But formik does not support file upload which was giving me a tough time so I changed it to custom form but I will work on it and improvise.